lnbits/db.py

```diff
@@ -126,6 +120,24 @@ def timestamp_placeholder(cls):
         else:
             return "?"
 
+    @classmethod
+    def truncate_date(cls, col: str, trunc: DateTrunc) -> str:
+        """
+        Removes all parts of the given date up until `trunc`.
+        Based on `date_trunc` postgres function
+        """
+        if DB_TYPE == SQLITE:
+            formats = {
+                "hour": "%Y-%m-%d %H:00:00",
+                "day": "%Y-%m-%d 00:00:00",
+                "month": "%Y-%m-01 00:00:00",
+            }
+            if trunc in formats:
+                return f"strftime('{formats[trunc]}', {col}, 'unixepoch')"
+        elif trunc in ("day", "hour", "month"):
+            return f"date_trunc('{trunc}', {col})"
+        raise ValueError(f"Invalid truncation: {trunc}")
```

Comment:
What exactly does this do? In sqlite, we used to store unix timestamps whereas in postgres, we used the `date` type. Does this affect anything? Would be nice to clarify.

Reply:
No, it just affects the way those timestamps are being used in the query.